### ISSUE TITLE
Legacy config download

### ISSUE THREAD

**Issue by troPeLcirE** [thin-edge/thin-edge.io#3284]
In my undestanding, when downloading configuration from Cumulocity, agent received smartrest 524 request and forwards it to te/main/device/cmd/config_update. However when using Legacy configuration smartrest id 521 is used and nothing is forwarded to te/. Is it possible to make smartrest 521 be handled as 524?

**Comment by reubenmiller** [comment]
@troPeLcirE Do you mind sharing the reasoning behind wanting ot support the legacy configuration file types? Do you just have existing components which make use of it, and don't want to break them?

I did a quick experiment (by only changing 2 lines of code), and thin-edge.io might be able to support this natively when it parsing the incoming operation (we use the full json structure now, and not the SmartREST 2.0 static templates). Below is the proposed solution, please give some feed back if this sounds ok.

**Proposed solution**

Use a default type of "legacy" if the `.type` property is not provided in either the `c8y_DownloadConfigFile` or `c8y_UploadConfigFile` fragment in the Cumulocity operations.

Users can then control file path associated with the "legacy" file type by the standard `tedge-configuration-plugin.toml` file,

**file: /etc/tedge/plugins/tedge-configuration-plugin.toml**

```toml
files = [
    { path = '/etc/tedge/tedge.toml', user = 'tedge', group = 'tedge', mode = 0o444 },
    { path = '/etc/tedge/system.toml', type = 'system.toml', user = 'tedge', group = 'tedge', mode = 0o444 },

    # Legacy configuration file type --
    { path = '/etc/legacy.txt', type = 'legacy', user = 'tedge', group = 'tedge', mode = 0o644 },
]
```

**Comment by reubenmiller** [comment]
> The reason I am asking for this is indeed that is the way our current interface sends configuration update request. Our custom interface is less flexible than default Cumulocity one and does not allow to choose type of configuration. This interface is available for several products (not managed by thin-edge) and I do not control changes on this part. And yes I will be very happy to see those 2 lines.

Ok, great. Then we'll proceed with the proposed solution of using the reserved "legacy" type for untyped configurations.

**Comment by reubenmiller** [comment]
@troPeLcirE We're just discussing legacy configuration file upload. Are you ok with the following limitation?

thin-edge.io will not support the configuration file upload (from the device to the cloud) using the legacy non-type based api, (as described here: https://cumulocity.com/docs/device-integration/fragment-library/#install-legacy-configuration). Instead the user can request the file using the type-based api (e.g. selecting the "default" configuration and clicking the "Retrieve snapshot")

We believe this will provide a much better user experience as it pushes them to use the new type-based interface, which thin-edge.io supports out of the box, and they can request the `default` type (we renamed it from `legacy`) using the same mechanism as any other configuration type. It is only the downloading of the legacy (non-type based configuration) which is mapped to the new `default` type (and defined in the `tedge-configuration-plugin.toml`).

**Comment by troPeLcirE** [comment]
Your argument looks valid. My plan is to support both Legacy and typed configuration download and push user interface team to allow typed configuration management as soon as possible.

**Comment by reubenmiller** [comment]
@troPeLcirE The PR has now been merged and will be included in the next release 1.4.0 which is scheduled for next week.

If you want to check it in the meantime, you can install the main channel version (e.g. a version built after we merge into the main branch), using the install.sh script:

```sh
wget -O - thin-edge.io/install.sh | sh -s -- --channel main
```

Though be aware it will configure the [tedge-main](https://cloudsmith.io/~thinedge/repos/tedge-main/packages/) linux repositories instead of the official release repo, [tedge-release](https://cloudsmith.io/~thinedge/repos/tedge-release/packages/). So you will actively have to remove the tedge-main repos if you want to switch back to the official releases.
